### ISSUE TITLE
MultiBinding provides error values to converter

### ISSUE THREAD

**Issue by jp2masa** [AvaloniaUI/Avalonia#2623]
I already investigated the problem a bit, but I'm not familiar with the `Binding` code nor with `WeakReference`s, which I believe are part of the issue.

In my code the first binding works, and the second one contains a binding notification of an error. In the repro, both contain error notifications. I also debugged a bit and found that the binding target (contained in a `WeakReference`) was `null`, so it may be a problem with losing those references.

The notification error message is this:

```
{{Error: Avalonia.Data.Core.MarkupBindingChainException: Null value in expression ''.}}
```

Any help would be appreciated.

<details>
<summary>Repro</summary>

In `ControlCatalog`, `ProgressBarPage.xaml`, replace:

```xaml
        <ProgressBar Value="{Binding #vprogress.Value}" Orientation="Vertical" />
```

with:

```xaml
          <ProgressBar Orientation="Vertical" Maximum="600">
            <ProgressBar.Value>
              <MultiBinding Converter="{x:Static pages:MyConverter.Instance}">
                <Binding />
                <Binding Path="Bounds.Height" RelativeSource="{RelativeSource Mode=FindAncestor, AncestorType=UserControl}" />
              </MultiBinding>
            </ProgressBar.Value>
          </ProgressBar>
```

and add namespace import:

```xaml
<UserControl ...
             xmlns:pages="clr-namespace:ControlCatalog.Pages"
             ...>
```

Add a new class named `MyConverter` to the `Pages` folder:

```cs
using System;
using System.Collections.Generic;
using System.Globalization;

using Avalonia.Data.Converters;

namespace ControlCatalog.Pages
{
    internal sealed class MyConverter : IMultiValueConverter
    {
        public static readonly MyConverter Instance = new MyConverter();

        public object Convert(IList<object> values, Type targetType, object parameter, CultureInfo culture)
        {
            return values[1];
        }
    }
}
```
</details>

**Comment by Jmales** [comment]
Was this solved? I seem to have an issue related to this one, if not the same.

**Comment by grokys** [comment]
This happens because when the `Binding` with a `RelativeSource` is set up, the control is still not part of the visual tree, so the binding returns "unset". `RelativeSource=FindAncestor` looks in the visual tree for the ancestor.

It's a similar problem to #8672, but not exactly the same. You can actually fix this by using the `$parent[UserControl]` syntax which looks in the logical tree, which is set up by the point the binding is initiated; however `DataContext` is still `null` at this point, so the first binding still produces "unset".

Is using the logical tree an option for you?

**Comment by grokys** [comment]
I've taken a look at WPF and it has a [bunch of logic](https://github.com/dotnet/wpf/blob/89d172db0b7a192de720c6cfba5e28a1e7d46123/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Data/BindingExpression.cs#L523) which it uses to determine when a `MultiBinding` can be attached in order to start producing values. Avalonia doesn't have this much information as our bindings are based on Observables, so I'm not sure we can replicate the behavior there.

**Comment by timunie** [comment]
works for me after #13970 which is available in nightly. If that still happens to you, ping me.

**Comment by LZD121** [comment]
I am using Avalonia 11.0.10, the unset value issue still happens to me.
